homeassistant/components/heatzy/climate.py

```diff
@@ -0,0 +1,295 @@
+
+    def __init__(
+        self, coordinator: HeatzyDataUpdateCoordinator, unique_id: str
+    ) -> None:
+        """Init."""
+        super().__init__(coordinator)
+        self._attr_unique_id = unique_id
+        self._attr_name = coordinator.data[unique_id][CONF_ALIAS]
+        self._attr_device_info = DeviceInfo(
+            identifiers={(DOMAIN, unique_id)},
+            name=coordinator.data[unique_id][CONF_ALIAS],
+            manufacturer=DOMAIN,
+            sw_version=coordinator.data[unique_id].get(CONF_VERSION),
+            model=coordinator.data[unique_id].get(CONF_MODEL),
+        )
+
+    @property
+    def hvac_mode(self) -> HVACMode:
+        """Return hvac operation ie. heat, cool mode."""
+        if self.preset_mode == PRESET_NONE:
+            return HVACMode.OFF
+        return HVACMode.HEAT
+
+    async def async_set_hvac_mode(self, hvac_mode: HVACMode) -> None:
+        """Set new hvac mode."""
+        if hvac_mode == HVACMode.OFF:
+            await self.async_turn_off()
+        elif hvac_mode == HVACMode.HEAT:
+            await self.async_turn_on()
+
+    async def async_turn_on(self) -> None:
+        """Turn device on."""
+        await self.async_set_preset_mode(PRESET_COMFORT)
+
+    async def async_turn_off(self) -> None:
+        """Turn device off."""
+        await self.async_set_preset_mode(PRESET_NONE)
+
+
+class HeatzyPiloteV1Thermostat(HeatzyThermostat):
+    """Heaty Pilote v1."""
+
+    HEATZY_TO_HA_STATE = {
+        "\u8212\u9002": PRESET_COMFORT,
+        "\u7ecf\u6d4e": PRESET_ECO,
+        "\u89e3\u51bb": PRESET_AWAY,
+        "\u505c\u6b62": PRESET_NONE,
+    }
+    HA_TO_HEATZY_STATE = {
+        PRESET_COMFORT: [1, 1, 0],
+        PRESET_ECO: [1, 1, 1],
+        PRESET_AWAY: [1, 1, 2],
+        PRESET_NONE: [1, 1, 3],
+    }
+
+    @property
+    def preset_mode(self) -> str | None:
+        """Return the current preset mode, e.g., home, away, temp."""
+        return self.HEATZY_TO_HA_STATE.get(
+            self.coordinator.data[self.unique_id].get(CONF_ATTR, {}).get(CONF_MODE)
+        )
+
+    async def async_set_preset_mode(self, preset_mode: str) -> None:
+        """Set new preset mode."""
+        try:
+            await self.coordinator.api.async_control_device(
+                self.unique_id,
+                {"raw": self.HA_TO_HEATZY_STATE.get(preset_mode)},
+            )
+            await self.coordinator.async_request_refresh()
+        except HeatzyException as error:
+            _LOGGER.error("Set preset mode (%s) %s (%s)", preset_mode, error, self.name)
+
+
+class HeatzyPiloteV2Thermostat(HeatzyThermostat):
+    """Heaty Pilote v2."""
+
+    HEATZY_TO_HA_STATE = {
+        "cft": PRESET_COMFORT,
+        "eco": PRESET_ECO,
+        "fro": PRESET_AWAY,
+        "stop": PRESET_NONE,
+    }
+
+    HA_TO_HEATZY_STATE = {
+        PRESET_COMFORT: "cft",
+        PRESET_ECO: "eco",
+        PRESET_AWAY: "fro",
+        PRESET_NONE: "stop",
+    }
```

Comment:
To keep things DRY/SSOT, this could be computed with a dictcomp from the other dict for brevity?
```suggestion
    HA_TO_HEATZY_STATE = {preset: name for (name, preset) in HEATZY_TO_HA_STATE.items()}
```